### PR TITLE
feat:default respondent org policy with solicitor claimant reference

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java [hunk 1]
```diff
@@ -190,6 +209,9 @@ private CallbackResponse submitClaim(CallbackParams callbackParams) {
         dataBuilder.allocatedTrack(getAllocatedTrack(caseData.getClaimValue().toPounds(), caseData.getClaimType()));
         dataBuilder.businessProcess(BusinessProcess.ready(CREATE_CLAIM));
 
+        if (caseData.getRespondent1OrgRegistered() == NO) {
+            dataBuilder.respondent1OrganisationPolicy(null);
+        }
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java:214]: what is this code doing?

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandlerTest.java [hunk 2]
```diff
@@ -174,6 +174,39 @@ void shouldReturnNoError_whenSoleTraderDateOfBirthIsInThePast() {
         }
     }
 
+    @Nested
+    class MidEventIsRespondentRepresentedCallback {
+
+        private final String pageId = "isResp1Represented";
+        private final CaseData.CaseDataBuilder caseDataBuilder =
+            CaseDataBuilder.builder().atStateClaimDraft().build().toBuilder();
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandlerTest.java]: surely a nicer way to do this that builder.builder.build.toBuilder 😅

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandlerTest.java [hunk 3]
```diff
@@ -174,6 +174,39 @@ void shouldReturnNoError_whenSoleTraderDateOfBirthIsInThePast() {
...
+        @Test
+        void shouldReturnRespOrgPolicy_whenRespondentRepresented() {
+            CaseData caseData = caseDataBuilder.respondent1Represented(YES).build();
+            CallbackParams params = callbackParamsOf(caseData, MID, pageId);
+
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getData())
+                .extracting("respondent1OrganisationPolicy.OrgPolicyReference")
+                .isNotNull()
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandlerTest.java]: is not null adds nothing here if we are asserting it's equal to something

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java (hunk 1 above)
Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/CreateClaimCallbackHandler.java:214]: respondent1OrganisationPolicy is set in casedata in previous page

**Review by pd-dudkiewicz — DISMISSED** [review]

**Review by pd-dudkiewicz — APPROVED** [review]

**Pull request closed without merging**